hookman/hookman_generator.py

```diff
@@ -711,6 +712,31 @@ def _generate_windows_body(hooks):
         f"    }}",
         f"",
         f"",
+        f"    HMODULE Load_DLL(const std::wstring& w_filename) {{",
+        f'        wchar_t *path_env = NULL;',
+        f'        rsize_t path_env_len = 0;',
+        f'        // Read PATH environment variable',
+        f'        _wdupenv_s(&path_env, &path_env_len, L"PATH");',
+        f'        std::wstring::size_type dir_name_size = w_filename.find_last_of(L"/\\\\");',
+        f'        int size = path_env_len + MAX_PATH;',
+        f'        wchar_t *new_path_env;',
+        f'        new_path_env = (wchar_t *)malloc(size);',
+        f'        rsize_t new_path_len = size;',
+        f'        // Get dir name from library(DLL) full path',
+        f'        auto dir_name = w_filename.substr(0, dir_name_size) + L";";',
+        f'        // Add dir name to PATH environment variable string',
+        f'        wcsncpy_s(new_path_env, new_path_len, dir_name.c_str(), dir_name_size+2);',
```

Comment:
Why `+2`?

Reply:
I removed this code. But it is need `+2` because I inserted an extra character `;` to concatenate the string in the PATH env variable.